Core/src/Vertexing/GaussianGridTrackDensity.cpp

```diff
@@ -93,71 +92,61 @@ Acts::GaussianGridTrackDensity<mainGridSize, trkGridSize>::addTrack(
   modifyMainGridWithTrackGrid(zBin, trkGrid, mainGrid, -1);
 }
 
-template <int mainGridSize, int trkGridSize>
-void Acts::GaussianGridTrackDensity<mainGridSize, trkGridSize>::
-    modifyMainGridWithTrackGrid(int zBin, const TrackGridVector& trkGrid,
-                                MainGridVector& mainGrid,
-                                int modifyModeSign) const {
-  int width = (trkGridSize - 1) / 2;
+void GaussianGridTrackDensity::modifyMainGridWithTrackGrid(
+    int zBin, const TrackGridVector& trkGrid, MainGridVector& mainGrid,
+    int modifyModeSign) const {
+  int width = (m_cfg.trkGridSize - 1) / 2;
   // Overlap left
   int leftOL = zBin - width;
   // Overlap right
-  int rightOL = zBin + width - mainGridSize + 1;
+  int rightOL = zBin + width - m_cfg.mainGridSize + 1;
   if (leftOL < 0) {
-    int totalTrkSize = trkGridSize + leftOL;
+    int totalTrkSize = m_cfg.trkGridSize + leftOL;
     mainGrid.segment(0, totalTrkSize) +=
         modifyModeSign * trkGrid.segment(-leftOL, totalTrkSize);
     return;
   }
   if (rightOL > 0) {
-    int totalTrkSize = trkGridSize - rightOL;
-    mainGrid.segment(mainGridSize - totalTrkSize, totalTrkSize) +=
+    int totalTrkSize = m_cfg.trkGridSize - rightOL;
+    mainGrid.segment(m_cfg.mainGridSize - totalTrkSize, totalTrkSize) +=
         modifyModeSign * trkGrid.segment(0, totalTrkSize);
     return;
   }
 
-  mainGrid.segment(zBin - width, trkGridSize) += modifyModeSign * trkGrid;
+  mainGrid.segment(zBin - width, m_cfg.trkGridSize) += modifyModeSign * trkGrid;
 }
 
-template <int mainGridSize, int trkGridSize>
-typename Acts::GaussianGridTrackDensity<mainGridSize,
-                                        trkGridSize>::TrackGridVector
-Acts::GaussianGridTrackDensity<mainGridSize, trkGridSize>::createTrackGrid(
-    float d0, float distCtrZ, const Acts::SquareMatrix2& cov) const {
-  TrackGridVector trackGrid(TrackGridVector::Zero());
-  float floorHalfTrkGridSize = static_cast<float>(trkGridSize) / 2 - 0.5f;
+GaussianGridTrackDensity::TrackGridVector
+GaussianGridTrackDensity::createTrackGrid(float d0, float distCtrZ,
+                                          const SquareMatrix2& cov) const {
+  TrackGridVector trackGrid(TrackGridVector::Zero(m_cfg.trkGridSize));
+  float floorHalfTrkGridSize = static_cast<float>(m_cfg.trkGridSize) / 2 - 0.5f;
 
   // Loop over columns
-  for (int j = 0; j < trkGridSize; j++) {
+  for (int j = 0; j < m_cfg.trkGridSize; j++) {
     float z = (j - floorHalfTrkGridSize) * m_cfg.binSize;
     trackGrid(j) = normal2D(-d0, z - distCtrZ, cov);
   }
   return trackGrid;
 }
 
-template <int mainGridSize, int trkGridSize>
-Acts::Result<float>
-Acts::GaussianGridTrackDensity<mainGridSize, trkGridSize>::estimateSeedWidth(
+Result<float> GaussianGridTrackDensity::estimateSeedWidth(
     MainGridVector& mainGrid, float maxZ) const {
-  if (mainGrid == MainGridVector::Zero()) {
+  if (mainGrid.isZero(0)) {
```

Comment:
```suggestion
  if (mainGrid.isZero()) {
```

Reply:
I faintly remember this was not doing the right thing without explicit precision, but I'll try.

Reply:
oh right `isZero()` is actually checking against an epsilong. w-t-f

Reply:
Seems to be ok.